src/sqlite3gen.cpp

```diff
@@ -224,221 +505,321 @@ struct SqlStmt {
-      ":raisable,"
-      ":kind,"
-      ":bodystart,"
-      ":bodyend,"
-      ":id_bodyfile,"
-      ":id_file,"
-      ":line,"
-      ":column,"
-      ":detaileddescription,"
-      ":briefdescription,"
-      ":inbodydescription"
-    ")"
-    ,NULL
+SqlStmt memberdef_exists={
+  "SELECT EXISTS (SELECT * FROM memberdef WHERE rowid = :rowid)"
+  ,NULL
+};
+
+SqlStmt memberdef_incomplete={
+  "SELECT EXISTS ("
+    "SELECT * FROM memberdef WHERE "
+    "rowid = :rowid AND inline != 2 AND inline != :new_inline"
+  ")"
+  ,NULL
+};
+
+SqlStmt memberdef_insert={
+  "INSERT INTO memberdef "
+  "("
+    "rowid,"
+    "name,"
+    "definition,"
+    "type,"
+    "argsstring,"
+    "scope,"
+    "initializer,"
+    "bitfield,"
+    "read,"
+    "write,"
+    "prot,"
+    "static,"
+    "const,"
+    "explicit,"
+    "inline,"
+    "final,"
+    "sealed,"
+    "new,"
+    "optional,"
+    "required,"
+    "volatile,"
+    "virt,"
+    "mutable,"
+    "initonly,"
+    "attribute,"
+    "property,"
+    "readonly,"
+    "bound,"
+    "constrained,"
+    "transient,"
+    "maybevoid,"
+    "maybedefault,"
+    "maybeambiguous,"
+    "readable,"
+    "writable,"
+    "gettable,"
+    "protectedsettable,"
+    "protectedgettable,"
+    "settable,"
+    "privatesettable,"
+    "privategettable,"
+    "accessor,"
+    "addable,"
+    "removable,"
+    "raisable,"
+    "kind,"
+    "bodystart,"
+    "bodyend,"
+    "bodyfile_id,"
+    "file_id,"
+    "line,"
+    "column,"
+    "detaileddescription,"
+    "briefdescription,"
+    "inbodydescription"
+  ")"
+  "VALUES "
+  "("
+    ":rowid,"
+    ":name,"
+    ":definition,"
+    ":type,"
+    ":argsstring,"
+    ":scope,"
+    ":initializer,"
+    ":bitfield,"
+    ":read,"
+    ":write,"
+    ":prot,"
+    ":static,"
+    ":const,"
+    ":explicit,"
+    ":inline,"
+    ":final,"
+    ":sealed,"
+    ":new,"
+    ":optional,"
+    ":required,"
+    ":volatile,"
+    ":virt,"
+    ":mutable,"
+    ":initonly,"
+    ":attribute,"
+    ":property,"
+    ":readonly,"
+    ":bound,"
+    ":constrained,"
+    ":transient,"
+    ":maybevoid,"
+    ":maybedefault,"
+    ":maybeambiguous,"
+    ":readable,"
+    ":writable,"
+    ":gettable,"
+    ":protectedsettable,"
+    ":protectedgettable,"
+    ":settable,"
+    ":privatesettable,"
+    ":privategettable,"
+    ":accessor,"
+    ":addable,"
+    ":removable,"
+    ":raisable,"
+    ":kind,"
+    ":bodystart,"
+    ":bodyend,"
+    ":bodyfile_id,"
+    ":file_id,"
+    ":line,"
+    ":column,"
+    ":detaileddescription,"
+    ":briefdescription,"
+    ":inbodydescription"
+  ")"
+  ,NULL
+};
+/*
+We have a slightly different need than the XML here. The XML can have two
+memberdef nodes with the same refid to document the declaration and the
+definition. This doesn't play very nice with a referential model. It isn't a
+big issue if only one is documented, but in case both are, we'll fall back on
+this kludge to combine them in a single row...
```

Comment:
Not sure common, but there's a chance both a declaration and a definition have their own documenting comments. This isn't a big problem in XML, but in SQL we need to combine the entries for referential integrity. There are a few ways to proceed:

- have separate columns for saving declaration/definition descriptions
- intentionally ignore one of the descriptions (i.e., only take the definition)
- implicitly ignore one (i.e., keep only the first-set or last-set)
- use some heuristic to combine the descriptions

For now, at least, I've chosen the latter. If encountered, the description fields will end up looking something like `Declaration: <decl_desc> Definition: <def_desc>`. I may need to polish this a little if we retain this practice.